File: examples/spring-boot-maven-plugin/src/main/java/community/flock/wirespec/examples/app/user/UserConverter.java
package community.flock.wirespec.examples.app.user;

import community.flock.wirespec.examples.app.common.Converter;
import community.flock.wirespec.generated.java.UserDto;
import org.springframework.stereotype.Component;

@Component
public class UserConverter implements Converter<User, UserDto> {
    @Override
    public User internalize(UserDto userDto) {
        return new User(userDto.name());
    }

    @Override
    public UserDto externalize(User user) {
        return new UserDto(user.name());
    }
}

File: examples/wirespec-gradle-plugin-ktor/src/main/kotlin/community/flock/wirespec/examples/app/user/UserConverter.kt
package community.flock.wirespec.examples.app.user

import community.flock.wirespec.examples.app.common.Converter
import community.flock.wirespec.generated.kotlin.UserDto

object UserConverter : Converter<User, UserDto> {
    override fun UserDto.internalize() = User(name = name)

    override fun User.externalize() = UserDto(name = name)

}
